### PR TITLE
DC/DC support for PCA10040

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

**Review by ioannisg — CHANGES_REQUESTED** [review]
LGTM.

But, actually, I am wondering: why do we need to have this K-config option at all? Couldn't we just select SOC_DCDC_NRF52X in Kconfig.board,or <board>_defkconfig?
Same for nRF52840_pca10056, where the DCDC is already selected in the same way.

I would understand the K-config option, if it were used by other (nonNordic( boards. But it is not, AFAICS

**Comment by cvinayak** [comment]
@ioannisg SOC_DCDC_NRF52X is hidden for DCDC feature's need of external components on the board, hence nRF52x SoC have them hidden only to be "select"-ed by boards (our DKs with visible BOARD_HAS_DCDC) that have the external components.

**Review by cvinayak — APPROVED** [review]

**Comment by ioannisg** [comment]
@cvinayak ,  I don't disagree on this. Just wondering; could SOC_DCDC_NRF52X be selected directly in the boards' K-config definition?

**Comment by carlescufi** [comment]
@ioannisg I think DC/DC is a compromise, and not everybody wants it enabled?

**Comment by ioannisg** [comment]
OK, so all we want is to allow users to be able to un-select it, from their prj.conf, even if they are using an "official" Nordic DK, right?

**Comment by carlescufi** [comment]
>OK, so all we want is to allow users to be able to un-select it, from their prj.conf, even if they are using an "official" Nordic DK, right?

Yes, correct

**Comment by ioannisg** [comment]
@carlescufi understood.
Then, it seems that BOARD_HAS_DCDC is also a "user" option. I am wondering if we could write something in the comment, inside the K-config option definition or prompt (e.g. "set to -n- to remove the DCDC feature"). Otherwise, it will look that it is solely an option describing hardware capabilities (and, clearly, it is not).

File: boards/arm/nrf52810_pca10040/Kconfig [hunk 1]
```diff
@@ -6,4 +6,9 @@
 
 if BOARD_NRF52810_PCA10040
 
+config BOARD_HAS_DCDC
```
Comment (nashif) [inline boards/arm/nrf52810_pca10040/Kconfig]: Kconfigs with _HAS_  should not be prompted, those should be in general hidden Kconfigs.  Also, where are those being used? You might want to change the name of the Kconfig and the prompt text .

**Review by nashif — DISMISSED** [review]

**Comment by carlescufi** [comment]
@ioannisg @nashif fixed all comments

File: boards/arm/nrf52810_pca10040/Kconfig [hunk 2]
```diff
@@ -6,4 +6,9 @@
 
 if BOARD_NRF52810_PCA10040
 
-endif # BOARD_NRF52810_PCA10040
+config BOARD_ENABLE_DCDC
+        bool "Enable DCDC mode"
+        select SOC_DCDC_NRF52X
+        default y
+
+ # BOARD_NRF52810_PCA10040
```
Comment (ioannisg) [inline boards/arm/nrf52810_pca10040/Kconfig]: Is there something wrong here, @carlescufi ?

Comment (ioannisg) [reply boards/arm/nrf52810_pca10040/Kconfig]: Looks in my diff that endif is missing

Comment (carlescufi) [reply boards/arm/nrf52810_pca10040/Kconfig]: Fixed, sorry.

**Review by ioannisg — APPROVED** [review]

**Pull request merged**